jobs/storage_test.groovy

```diff
@@ -0,0 +1,69 @@
+evaluate(new File("${WORKSPACE}/common.groovy"))
+
+name = "storage_backend_e2e"
+repoName = 'charts'
+
+job(name) {
+  description """
+    <p>Nightly Job runs the <a href="https://github.com/deis/workflow-e2e">e2e tests</a> against a <a href="https://github.com/deis/charts/tree/master/${defaults.workflowChart}">${defaults.workflowChart}</a> chart configured to GCS by default </p>
+  """.stripIndent().trim()
+
+  scm {
+    git {
+      remote {
+        github("deis/${repoName}")
+      }
+      branch('master')
+    }
+  }
+
+  logRotator {
+    daysToKeep defaults.daysToKeep
+  }
+  parameters {
+   stringParam('STORAGE_TYPE', 'gcs', "storage backend for helm chart, default is gcs")
+  }
+
+  triggers {
+    cron('@daily')
+  }
+
+  wrappers {
+   buildName('Storage_backed-${STORAGE_TYPE} #${BUILD_NUMBER}')
+    timeout {
+      absolute(30)
+      failBuild()
+    }
+    timestamps()
+    colorizeOutput 'xterm'
+    credentialsBinding {
+      string("GCLOUD_CREDENTIALS", "246d6550-569b-4925-8cda-e11a4f0d6803")
+      string("GCS_KEY_JSON", "71dd890b-e4ce-4483-97ca-a7c286c2381c")
+      string("AWS_SECRET_KEY","033f76e3-21b8-4bae-a7b3-7f43e07138d3")
+      string("AWS_ACCESS_KEY","02a5c84e-3248-4776-acc5-6b6a1fb24dfc")
+    }
+  }
+
+  steps {
+    shell '''
+      #!/usr/bin/env bash
+
+      set -eo pipefail
+
+      rand="${RANDOM}"
+      echo $rand
+      if [ ${STORAGE_TYPE} == "gcs" ]; then
+        TYPE="GCS"
+      else
+        TYPE="AWS"
+        export S3_REGION="us-west-1"
+      fi
+      export ${TYPE}_REGISTRY_BUCKET="deis-registry-${rand}"
```

Comment:
as mentioned above, we could eliminate the use of `rand` and just do something like this:
```
export ${TYPE}_REGISTRY_BUCKET="deis-registry-${BUILD_NUMBER}"
export ${TYPE}_BUILDER_BUCKET="deis-builder-${BUILD_NUMBER}"
export ${TYPE}_DATABASE_BUCKET="deis-database-${BUILD_NUMBER}"
```
I'd say we might also add `jenkins`, `${JOB_NAME}` and/or another such identifier to be explicitly clear it was created by ci...

PS here is a list of some of the env vars we get from the job DSL: https://github.com/jenkinsci/job-dsl-plugin/wiki/User-Power-Moves#access-the-jenkins-environment-variables